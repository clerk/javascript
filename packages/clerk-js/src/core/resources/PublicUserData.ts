import { deprecatedProperty } from '@clerk/shared';
import type { PublicUserData as IPublicUserData } from '@clerk/types';
import type { PublicUserDataJSON } from '@clerk/types';

export class PublicUserData implements IPublicUserData {
  firstName!: string | null;
  lastName!: string | null;
  /**
   * @deprecated  Use `imageUrl` instead.
   */
  profileImageUrl!: string;
  imageUrl!: string;
  hasImage!: boolean;
  identifier!: string;
  userId?: string;

  constructor(data: PublicUserDataJSON) {
    this.fromJSON(data);
  }

  protected fromJSON(data: PublicUserDataJSON | null): this {
    if (data) {
      this.firstName = data.first_name;
      this.lastName = data.last_name;
      this.profileImageUrl = data.profile_image_url;
      this.imageUrl = data.image_url;
      this.hasImage = data.has_image;
      this.identifier = data.identifier;
      this.userId = data.user_id;
    }

    return this;
  }
}

deprecatedProperty(PublicUserData, 'profileImageUrl', 'Use `imageUrl` instead.');
