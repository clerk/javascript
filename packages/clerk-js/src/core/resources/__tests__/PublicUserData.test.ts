import { PublicUserData } from '../PublicUserData';

describe('PublicUserData', () => {
  it('JSON.stringify returns the same object structure', () => {
    const pud = new PublicUserData({
      object: 'public_user_data',
      id: '123',
      first_name: 'John',
      last_name: 'Doe',
      image_url: 'https://example.com/image.jpg',
      has_image: true,
      identifier: 'john-doe',
      user_id: '123',
    });

    expect(JSON.stringify(pud)).toMatchSnapshot();
  });
});
