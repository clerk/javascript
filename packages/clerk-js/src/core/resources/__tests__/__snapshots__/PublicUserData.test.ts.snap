// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicUserData JSON.stringify returns the same object structure 1`] = `"{"firstName":"John","lastName":"Doe","imageUrl":"https://example.com/image.jpg","hasImage":true,"identifier":"john-doe","userId":"123"}"`;
